### DISCUSSION TITLE
Big data causes page jam

### DISCUSSION THREAD

**Discussion by Luyio** [fabricjs/fabric.js#8291] [Q&A]
Hi, I am working on a tracking function, which may have thousands of points to render. When I use groups and iterate to add data, the page will crash. Is there any good way to deal with it
`const group = new fabric.Group([], opts);
  graphs.map(res => group.add(res));
  group.addWithUpdate();`

**Comment by ShaMan123** [comment]
It should not crash
We have benchmarks for 10K objects in a group
Sounds something on your side
